src/python/pants/backend/codegen/utils.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from pants.backend.python.dependency_inference.module_mapper import (
+    ModuleProviderType,
+    ThirdPartyPythonModuleMapping,
+)
+from pants.engine.addresses import Address
+from pants.util.docutil import doc_url
+
+
+class MissingPythonCodegenRuntimeLibrary(Exception):
+    pass
+
+
+class AmbiguousPythonCodegenRuntimeLibrary(Exception):
+    pass
+
+
+def find_python_runtime_library_or_error(
+    module_mapping: ThirdPartyPythonModuleMapping,
+    codegen_address: Address,
+    runtime_library_module: str,
+    *,
+    recommended_requirement_name: str,
+    recommended_requirement_url: str,
+    disable_inference_option: str,
+) -> Address:
+    addresses = [
+        module_provider.addr
+        for module_provider in module_mapping.providers_for_module(
+            runtime_library_module, resolve=None
```

Comment:
In a followup, this will be the `python_resolve` of the `protobuf_source` target.